middleware/auth-utils/grant-manager.js

```diff
@@ -421,13 +421,19 @@ const fetch = (manager, handler, options, params) => {
 
     const req = getProtocol(options).request(options, (response) => {
       if (response.statusCode < 200 || response.statusCode > 299) {
-        return reject(response.statusCode + ':' + http.STATUS_CODES[ response.statusCode ]);
+        let response = '';
+        response.on('data', (d) => (response += d.toString()));
+        response.on('end', () => {
+          const message = response.statusCode + ':' + http.STATUS_CODES[ response.statusCode ] + ' ' + response;
+          reject(new Error(message));
+        });
+      } else {
+        let json = '';
+        response.on('data', (d) => (json += d.toString()));
+        response.on('end', () => {
```

Comment:
the starting of `else` block is identical to code in `if` block...  IMO, you should re-use it and put
`if (response.statusCode < 200 || response.statusCode > 299)`
to
`response.on('end', () => {`